datumaro/datumaro/components/project.py

```diff
@@ -315,6 +315,46 @@ def categories(self):
         return self._parent.categories()
 
 class Dataset(Extractor):
+    @classmethod
+    def from_iterable(cls, iterable, categories=None):
+        """Generation of Dataset from iterable object
+
+        Args:
+            iterable: Iterable object contains DatasetItems
+            categories (dict, optional): Dict of categories of datasets. Defaults to {}.
+
+        Returns:
+            Dataset: Dataset object
+        """
+
+        # Configuration of categories
```

Comment:
Please read about commenting your code https://www.elegantthemes.com/blog/wordpress/how-to-comment-your-code-like-a-pro-best-practices-and-good-habits

There is no need in commenting each line.